File: automated_email.js
const axios = require("axios");
const nodemailer = require("nodemailer");
const mysql = require("mysql");

async function getTemperature() {
  const city = "Tokyo";
  const url = `https://wttr.in/${city}?format=%t`;

  try {
    const response = await axios.get(url);
    return response.data.trim();
  } catch (error) {
    console.error("Error fetching temperature");
    return null;
  }
}

async function sendEmail(temperature, recipient, recipientName) {
  const email = "my_email_address";
  const password = "my_gmail_auth";

  const transporter = nodemailer.createTransport({
    host: "smtp.gmail.com",
    port: 465,
    secure: true,
    auth: {
      user: email,
      pass: password,
    },
  });

  const mailOptions = {
    from: email,
    to: recipient,
    subject: "Automated Email",
    text: `Hi ${recipientName}, how are you? Today's temperature in Japan is ${temperature}. I will come to NY soon.`,
  };

  try {
    await transporter.sendMail(mailOptions);
    console.log("Email sent successfully!");
  } catch (error) {
    console.error(`Error sending email: ${error}`);
  }
}

function getEmailData() {
  return new Promise((resolve, reject) => {
    const connection = mysql.createConnection({
      host: "localhost",
      user: "my_sql_username",
      password: "my_sql_pass",
      database: "export_test_database",
    });

    connection.connect();

    connection.query("SELECT Users, `E-mail` FROM Email_database", (error, results, fields) => {
      connection.end();

      if (error) {
        reject(error);
      } else {
        resolve(results);
      }
    });
  });
}

async function job() {
  const temperature = await getTemperature();
  if (temperature !== null) {
    const emailData = await getEmailData();
    for (const row of emailData) {
      await sendEmail(temperature, row["E-mail"], row.Users);
    }
  }
}

// Call the job function without scheduling
job();

File: automated_email.py
import smtplib
import requests
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
import mysql.connector

def get_temperature():
    city = "Tokyo"
    url = f"https://wttr.in/{city}?format=%t"
    
    response = requests.get(url)
    if response.status_code == 200:
        temperature = response.text.strip()
        return temperature
    else:
        print("Error fetching temperature")
        return None

def send_email(temperature, recipient, recipient_name):
    email = "my_email_address"
    password = "auth_pass"

    msg = MIMEMultipart()
    msg["From"] = email
    msg["To"] = recipient
    msg["Subject"] = "Automated Email"

    body = f"Hi {recipient_name}, how are you? Today's temperature in Japan is {temperature}. I will come to NY soon."
    msg.attach(MIMEText(body, "plain"))

    try:
        server = smtplib.SMTP_SSL("smtp.gmail.com", 465)
        server.login(email, password)
        server.sendmail(email, recipient, msg.as_string())
        server.quit()
        print("Email sent successfully!")
    except Exception as e:
        print(f"Error sending email: {e}")

def get_email_data():
    connection = mysql.connector.connect(
        host="localhost",
        user="my_sql_username",
        password="my_sql_pass",
        database="export_test_database"
    )

    cursor = connection.cursor()
    cursor.execute("SELECT Users, `E-mail` FROM Email_database")
    email_data = cursor.fetchall()
    cursor.close()
    connection.close()

    return email_data

def job():
    temperature = get_temperature()
    if temperature is not None:
        email_data = get_email_data()
        for recipient_name, recipient_email in email_data:
            send_email(temperature, recipient_email, recipient_name)

# Call the job function without scheduling
job()
